docs/explanations/type_validators.md

```diff
@@ -3,44 +3,44 @@
 ## Requirement
 
 Blueapi takes the parameters of a plan and internally creates a [pydantic](https://docs.pydantic.dev/) model for future validation e.g.
-``` 
-    def my_plan(a: int, b: str = "b") -> Plan
-        ...
+```python 
+def my_plan(a: int, b: str = "b") -> Plan
+    ...
 
-    # Internally becomes something like
+# Internally becomes something like
 
-    class MyPlanModel(BaseModel):
-        a: int
-        b: str = "b"
+class MyPlanModel(BaseModel):
+    a: int
+    b: str = "b"
```

Comment:
Is this repeating what is explained in plans.md?

Reply:
Probably does need a bit of consolidation, my read is that plans.md explains which parts of blueapi plans are special and why, and type_validators.md explains the very technical implementation of plan parameter parsing.

Reply:
I think this is just repeating what is in lifecycle.md. It even includes the same json example, I'd really like to see one removed or the difference between the two explained.

Reply:
So the idea is that `lifecycle.md` is a general overview of how blueapi stores, retrieves and runs plans whereas this is a deep dive into the special handling for substituting devices for strings. Do you have any suggestions as to what we can shuffle around to make that clearer?